### DISCUSSION TITLE
initialiser un dépôt Git sur un serveur local

### DISCUSSION THREAD

**Discussion by PhilippeD31** [ddt-adl-sig/dev#3] [Git & Github]
Si on veut collaborer en interne avec Git, sans créer un dépôt sur internet, on peut créer un *dépôt sur un serveur local*.

Voici un exemple avec une branche Git principale appelée "*main*"  
et un dossier réseau : \\\monserveur\dossier\git

### 1. Initialiser le projet Git (si ce n'est pas déjà fait)
Ouvrir un terminal Git CMD et exécuter les commandes :
```
cd C:\chemin\vers\mon\projet
git init
git add .
```

### 2. Créer le dépôt Git sur le serveur local
! Le dossier \\monserveur\dossier\git doit être vide.
```
cd C:\chemin\vers\mon\projet
pushd \\monserveur\dossier\git
git init --bare
popd
git remote add origin \\monserveur\dossier\git
git push origin main
```

**Comment by loicbcn** [comment]
Obtenir un export de  l'historique git
``` batch
git --no-pager log --graph --pretty=format:"%C(yellow)%h%x09%Creset%C(cyan)%C(bold)%ad%Creset %C(yellow)%cn%Creset  %C(green)%Creset %s" --date=iso
```
